docs/symfony-messenger.md

```diff
@@ -0,0 +1,72 @@
+# Using async Symfony Messenger
+
+Add new service to the `compose.yaml`:
+```yaml
+  php-worker:
+    image: ${IMAGES_PREFIX:-}app-php-worker
+    restart: unless-stopped
+    environment:
+      - RUN_MIGRATIONS=false
+    healthcheck:
+      disable: true
+    depends_on:
+      - php
+```
+
+Add new services to the `compose.override.yaml`:
+```yaml
+  php-worker:
+    profiles:
+      - donotstart
+
+  php-worker-async:
+    scale: 2
+    extends:
+      file: compose.yaml
+      service: php-worker
+    image: ${IMAGES_PREFIX:-}app-php-worker-async
+    build:
+      context: .
+      target: frankenphp_dev
+    command: ['bin/console', 'messenger:consume', 'async', '-vv', '--time-limit=60', '--limit=10', '--memory-limit=128M']
+    volumes:
+      - ./:/app
+      - /app/var/
+    depends_on:
+      php:
+        condition: service_healthy
+```
+
+Two instances of `php-worker-async` will start after `php` container which does installation of Symfony. They will share app folder because of `- ./:/app` in volumes configuration. `- /app/var/` defines that every container will have its own and separate `/app/var/` folder, [note missing `:`](https://stackoverflow.com/questions/46166304/docker-compose-volumes-without-colon).
+
+To add additional workers just copy `php-worker-async` service and replace every usage of the `async` in the new service with appropriate value for the new worker.
+
+Add new services to the `compose.prod.yaml`:
+```yaml
+  php-worker:
+    profiles:
+      - donotstart
+
+  php-worker-async:
+    scale: 2
+    extends:
+      file: compose.yaml
+      service: php-worker
+    image: ${IMAGES_PREFIX:-}app-php-worker-async
+    build:
+      context: .
+      target: frankenphp_prod
+    command: ['bin/console', 'messenger:consume', 'async', '-vv', '--time-limit=60', '--limit=10', '--memory-limit=128M']
+    depends_on:
+      php:
+        condition: service_healthy
+```
+
+Apply the following changes to the `frankenphp/docker-entrypoint.sh`:
+```patch
+-	if grep -q ^DATABASE_URL= .env; then
++	if grep -q ^DATABASE_URL= .env && [ "${RUN_MIGRATIONS:-true}" = "true" ]; then
```

Comment:
So basically this won't wait anymore for the database to be available?

Reply:
@norkunas, I agree with you. A similar proposal [was made above](https://github.com/dunglas/symfony-docker/pull/681#issuecomment-2454054046).

[The author refers](https://github.com/dunglas/symfony-docker/pull/681#issuecomment-2468769518) to the fact that the service will only be launched after a healthy `php` service. And this is true.

Although I would abandon the dependency on the `php` service (so that it would be possible to launch the consumer separately from or earlier than the main service), by enabling the wait for the database to be ready. However, in this case, consumers will be launched before the migrations are applied. So for now I agree with the author on the issue of waiting for the database.